### DISCUSSION TITLE
Document internals of DRF

### DISCUSSION THREAD

**Discussion by JerzySpendel** [encode/django-rest-framework#7804] [Ideas & Suggestions]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [ ] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

What do you think about documenting internals of the DRF? I've spent quite a lot of time digging in the implementations of different fields/serializers classes to understand what's the information flow between different methods (this is barely covered in the current API documentation). At the end I was satisfied because I got better understanding how thing works under the hood, but it was really time-consuming. What would you think if started working on documenting the internals of DRF?

**Comment by rpkilby** [comment]
Hi @JerzySpendel. What kind of documentation are you thinking about? Docstrings for classes/functions, or separate topics that discuss the overall organization?

I would suggest taking a stab at something limited in scope, get feedback, and then we can work from that.

**Comment by krdeepak** [comment]
I agree. I have been going through code and documentation a lot for past few days and having knowledge of internals will really be helpful. I will start working on a some of it soon. To being with, I will take few particular customisations, implement it and then show how it interacts with internal implementation.
